### ISSUE TITLE
Help with TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by yasmeencole** [nss-day-cohort-46/How-to-Ask-for-Help#29]
#### Have you searched the other issue tickets? If not, do that first. 
- [ x] Yes
- If No, stop and search.

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
 

- [x ] Added label to specific project/assignment that you need help with
- [ x] Added to `c46 Help-Thread` project board

#### REQUIRED Describe your issue:
>>

#### REQUIRED List any error messages that appear:
OpenCart.js:23 Uncaught (in promise) TypeError: Cannot read property 'name' of undefined
    at render (OpenCart.js:23)
    at OpenCart (OpenCart.js:12)
    at OpenCart.js:54

#### REQUIRED What have you googled? 

- [ ] 1. http://net-informations.com/js/iq/unerror.htm
- [ ] 2. https://trungk18.com/experience/uncaught-type-error/
- [ ] 3. https://www.pluralsight.com/guides/use-es6-arrow-functions-to-resolve-%22typeerror:-cannot-read-property-'lessyour-property-namegreater'-of-undefined%22

#### REQUIRED: What have you tried? Tell us everything you have tried. 
>> I have tried researching the error, using the debugger, and getting help from teammate. 

#### REQUIRED: 
**Your link:** https://github.com/nss-day-cohort-46/sprinkles-of-joy-c3pie-o/pull/20/files

#### Code Snippet

```

  for (const product of productsInCart) {
    cartHTML += `
      <div class="cart">
        <p>${product.name}</p>
        <p>$${product.price.toFixed(2)}</p>
      </div>
    `
    totalCost += product.price
  }
```

**Comment by jisie** [comment]
@yasmeencole This means that instead adding a product object to your array that represents your open cart `productsInCart`, you are adding undefined. Start debugging from when the user clicks `Add to cart`. Questions to consider:

1. Are you dispatching an event?
1. Does the event listener hear the event?
1. Does the name of the event match? 
1. If there is a data payload, can you access the data in the event listener callback function?
1. Are you finding the product object for the object that you want to add to the cart from you collection of products?
1. Is that object being added to your `productsInCart` array?

**Comment by jisie** [comment]
@yasmeencole Any luck with debugging?

**Comment by daprieu** [comment]
yes we just solved it!! Well Yasmeen did I just helped with debugging.

**Comment by jisie** [comment]
@yasmeencole @daprieu That's great!! Can you please add a comment describing how you resolved the issue and close this issue?

**Comment by yasmeencole** [comment]
I solved this issue with the help of @daprieu. One thing we noticed on OpenCart.js line 11, is that the export const OpenCart  was empty and was not getting the products to be added to the cart. Also, the export statement was not rendering productsInCart to the dom. After that we debugged Products.js where we determined that the eventlistener on line 28 was grabbing the value (productId) but we actually needed to be grabbing the property (addedProduct).
